python/ray/util/multiprocessing/pool.py

```diff
@@ -186,46 +187,65 @@ def _add_object_ref(self, object_ref):
         self._object_refs.append(object_ref)
         self._results.append(None)
 
-    def add_object_ref(self, object_ref):
+    def add_object_ref(self, object_ref, final=False):
         self._new_object_refs.put(object_ref)
+        if final:
```

Comment:
I think it might be clearer to leave the `final` conditional out of this function (and have the caller implement it directly). It doesn't hide that much complexity, and it would make the caller's code easier to follow, especially if the caller is putting [`ResultThread.END_SENTINEL`](https://github.com/ray-project/ray/pull/24117/files#r857976076) instead of `None`.

Reply:
Indeed, this has been fixed by the `END_SENTINEL` suggestion.